src/main/java/org/broadinstitute/hellbender/tools/exome/segmentation/PerformJointSegmentation.java

```diff
@@ -0,0 +1,126 @@
+            optional = false
+    )
+    protected String coverageFile;
+
+    @Argument(
+            doc = "Input file for tumor-sample ref/alt read counts at normal-sample heterozygous-SNP sites (output of GetHetCoverage tool).",
+            fullName = ExomeStandardArgumentDefinitions.TUMOR_ALLELIC_COUNTS_FILE_LONG_NAME,
+            shortName = ExomeStandardArgumentDefinitions.TUMOR_ALLELIC_COUNTS_FILE_SHORT_NAME,
+            optional = false
+    )
+    protected File snpCountsFile;
+
+    @Argument(
+            doc = "Initial number of hidden copy-ratio states",
+            fullName = INITIAL_NUM_COPY_RATIO_STATES_LONG_NAME,
+            shortName = INITIAL_NUM_COPY_RATIO_STATES_SHORT_NAME,
+            optional = false
+    )
+    protected int initialNumCRStates;
+
+    @Argument(
+            doc = "Initial number of hidden allele-fraction states",
+            fullName = INITIAL_NUM_ALLELE_FRACTION_STATES_LONG_NAME,
+            shortName = INITIAL_NUM_ALLELE_FRACTION_STATES_SHORT_NAME,
+            optional = false
+    )
+    protected int initialNumAFStates;
+
+    @Argument(
+            doc = "Input file for allelic-bias panel of normals.",
+            fullName = ExomeStandardArgumentDefinitions.ALLELIC_PON_FILE_LONG_NAME,
+            shortName = ExomeStandardArgumentDefinitions.ALLELIC_PON_FILE_SHORT_NAME,
+            optional = true
+    )
+    protected File allelicPoNFile;
+
+    @Argument(
+            doc = "Output file for copy-ratio segments.  Output is not logged.",
+            fullName = ExomeStandardArgumentDefinitions.SEGMENT_FILE_LONG_NAME,
+            shortName = ExomeStandardArgumentDefinitions.SEGMENT_FILE_SHORT_NAME,
+            optional = false
+    )
+    protected File outputSegmentsFile;
+
+    @Override
+    public Object doWork() {
+        final AllelicPanelOfNormals allelicPoN =
+                allelicPoNFile != null ? AllelicPanelOfNormals.read(allelicPoNFile) : AllelicPanelOfNormals.EMPTY_PON;
+        final AllelicCountCollection acc = new AllelicCountCollection(snpCountsFile);
+        final AlleleFractionSegmenter alleleFractionSegmenter = new AlleleFractionSegmenter(initialNumAFStates, acc, allelicPoN);
+
+        final ReadCountCollection counts;
+        try {
+            counts = ReadCountCollectionUtils.parse(new File(coverageFile));
+        } catch (final IOException ex) {
+            throw new UserException.BadInput("could not read input file");
+        }
+
+        final List<Double> coverage = Doubles.asList(counts.counts().getColumn(0));
+        final List<SimpleInterval> targets = counts.targets().stream().map(Target::getInterval).collect(Collectors.toList());
+        final CopyRatioSegmenter copyRatioSegmenter = new CopyRatioSegmenter(initialNumCRStates, targets, coverage);
+
+        final JointAFCRSegmenter jointSegmenter = JointAFCRSegmenter.createJointSegmenter(copyRatioSegmenter, alleleFractionSegmenter);
```

Comment:
Looks like the `CopyRatioSegmenter` constructor still takes in `final List<SimpleInterval> positions, final List<Double> data` rather than a `ReadCountCollection`?  OK if you chose not to take my second suggestion, but just wanted to make sure it was clear!